cifopy/charles/charles.py

```diff
@@ -108,9 +124,34 @@ def evolve(self, gens, select, crossover, mutate, co_p, mu_p, elitism):
             self.individuals = new_pop
  
             if self.optim == "max":
-                print(f'Best Individual: {max(self, key=attrgetter("fitness"))}')
+                if gen%50==0:
+                    print(f'Best Individual: {max(self, key=attrgetter("fitness"))} Gen: {gen}')
+ 
             elif self.optim == "min":
-                print(f'Best Individual: {min(self, key=attrgetter("fitness"))}')
+                if gen%50==0:
+                    print(f'Best Individual: {min(self, key=attrgetter("fitness"))} Gen: {gen}')
+    
+    def reshuffle(self):
+        n = len(self)
+        new_pop = []
+        if self.optim == "max":
+            new_pop.append(deepcopy(max(self.individuals, key=attrgetter("fitness"))))
+        elif self.optim == "min":
+            new_pop.append(deepcopy(min(self.individuals, key=attrgetter("fitness"))))
+        
+        
+        keep = sample(range(n), k=n//2)
+        #for how it's implemented the best could appear twice. Don't think it's a problem but we can avoid it
+        #half of the times you keep the best twice, could also be useful. write in report
+        for i in keep:
+            new_pop.append(deepcopy(self.individuals[i])) #to test, not sure
```

Comment:
Do we actually need to deep copy them? The individuals won't be used anymore anyway right?

Reply:
I did not want to make any mess with pointers and individuals, I think also Dave deepcopies often. Probably would work the same, but anyway few lines later self.individuals is overwritten by new_pop, so the old individuals will be discarded anyway.